app/src/main/java/it/quip/android/view/CircleTagTextView.java

```diff
@@ -0,0 +1,60 @@
+package it.quip.android.view;
+
+import android.text.SpannableStringBuilder;
+import android.text.TextPaint;
+import android.graphics.Color;
+import android.text.Spannable;
+
+import android.text.style.ClickableSpan;
+import android.util.Log;
+import android.view.View;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import it.quip.android.listener.CircleTagClicked;
+import it.quip.android.util.FormatUtil;
+
+public class CircleTagTextView {
```

Comment:
This can be generalized to be just ```TagTextSpanner``` where the ```Pattern``` is a parameter passed in such that we can pass in one for Circle patterns, Hashtag patterns, etc.